v2/protocol/http/protocol.go

```diff
@@ -118,6 +118,16 @@ func (p *Protocol) Request(ctx context.Context, m binding.Message) (binding.Mess
 	if err = WriteRequest(ctx, m, req, p.transformers); err != nil {
 		return nil, err
 	}
+
+	do := p.doOnce
+	if delay, tries := cecontext.LinearBackoffFrom(ctx); tries > 0 {
```

Comment:
I would change this method to `BackoffFrom` and return the `Backoff` struct, then have a switch for `BackoffStrategyLinear` or `BackoffStrategyNone`